lib/application/widgets/theme/memo_terminal.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:layoutr/common_layout.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/constants/images.dart' as images;
+import 'package:memo/application/constants/strings.dart' as strings;
+import 'package:memo/application/hooks/rich_text_field_controller_hook.dart';
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/utils/bottom_sheet.dart';
+import 'package:memo/application/widgets/theme/custom_button.dart';
+import 'package:memo/application/widgets/theme/rich_text_field.dart';
+import 'package:memo/application/widgets/theme/terminal_window.dart';
+
+/// A terminal-styled component that presents a `Memo` question and answer that can be updated.
+///
+/// Use [questionController] and [answerController] to control the Memo content being edited.
+class MemoTerminal extends HookConsumerWidget {
```

Comment:
This really look super-specific to the needs of your UI in the "Update Collection", as it's creeping with a lot of specific functionality, like "memoIndex" and "onRemove".

I would suggest to make it more specific to your use case and not to try and abstract it before needing it (because it's not really adaptable to be used in multiple places, cause it's filled with context-specific properties and behaviors).

Reply:
`TerminalWindow` does indeed make sense and it's really independent, but this widget doesn't seems to me like the case.

Reply:
I wasn't aiming for abstraction, but I agree with your point. What about rename to `UpdateMemoTerminal`?